utah-design-system-website/src/react/components/websiteContent/library/components/buttons/confirmationButton/ConfirmationButtonExampleRender.jsx

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import {
+  ConfirmationButton,
+  useBanner
+} from '@utahdts/utah-design-system';
+
+/** @typedef {import('../../../../../../../typedefs.d').ConfirmationButtonExamplePropsShape} ConfirmationButtonExamplePropsShape */
+
+/**
+ * @param {Object} props
+ * @param {import('use-immer').Updater<{props: ConfirmationButtonExamplePropsShape}>} props.setState
+ * @param {{props: ConfirmationButtonExamplePropsShape}} props.state
+ * @param {React.RefObject} props.innerRef
+ * @returns {JSX.Element}
+ */
+export function ConfirmationButtonExampleRender({
+  state: {
+    props: {
+      appearance,
+      isBusy,
+      className,
+      color,
+      isDisabled,
+      id,
+      size,
+      title,
+      promptChildren,
+      type,
+    },
+  },
+  innerRef,
+}) {
+  const showBanner = useBanner();
+  return (
+    <ConfirmationButton
+      appearance={appearance}
+      className={className}
+      color={color}
+      id={id}
+      innerRef={innerRef}
+      isBusy={isBusy}
+      isDisabled={isDisabled}
+      onClick={() => showBanner({ message: 'You have clicked the button.' })}
+      size={size}
+      type={type}
+      promptChildren={promptChildren}
+    >
+      {title || ''}
```

Comment:
maybe it would be more composable to have ConfirmationButtonInitial and ConfirmationButtonPrompt components?

Reply:
```
<ConfirmationButton>
  <InitialChildren>
    Delete the internet!
  </InitialChildren>
  <ConfirmationChildren>
   Are you sure? (Do you have backup?)
  </ConfirmationChildren>
...
  <IconForButton />
</ConfirmationButton>
```

The children use the context to determine if they themselves should show.

Reply:
This is the right way. It potentially creates too much for for the sprint. It also is not exactly the way <Button> works with icons.

Reply:
🦖  - I approve